src/ixdat/techniques/ec_ms.py

```diff
@@ -426,6 +429,156 @@ def siq_ecms_calibration_curve(
+        return impulse_response
+
+    def deconvolute_for_tspans(
+        self,
+        tspan_list,
+        t_zero_list,
+        impulse_response,
+        mol,
+        F_mol,
+        plot=True,
+        name=None,
+        t_bg=[-10, -1],
+        snr=7,
+        return_t_v_list=False,
+    ):
+        """
+        Loops though list of tspans and associated t_zero list to deconvolute using
+        the given impulse response (from model, but could also be from data) for
+        the molecule the impulse resp
+
+        The plots will only be meaningful if the current is calibrated.
+        #TODO: add error message
+        Args:
+            tspan_list (list): list of tspans to devonvolute data over
+            t_zero_list (list): zero point where the deconvolution starts
+                #TODO:need to understand if this is actually necessary
+            impulse_response (ImpulseResponse): impulse response object from model/data
+            mol (str): molecule
+            F_mol (CalPoint): spectro_inlets_calibration CalPoint object
+                #TODO: make it work with ixdat native?
+            plot (bool): will return a plot of each tspan
+            name (str): str to use for title in figure and saving
+            t_bg (tspan): tspan to be used as background IN RELATION TO t_zero.
+                Will be the same for each tspan. #TODO: add list option?
+            snr (int): signal-to-noise ratio used for Wiener deconvolution
+                (see grab_deconvoluted_current())
+            return_t_v_list (bool): Whether to return list of (time, deconvoluted
+                partial current densisty), Defaults to False
+
+        Return t_v_list (list): list of tuple of (time, deconvoluted partial current
+                   densisty) as returned from grab_deconvoluted_current()
+        """
+        from spectro_inlets_quantification import Calibration
+
+        t_v_list = []
+        for tspan, t_zero in zip(tspan_list, t_zero_list):
+            print(
+                "Now working on tspan {}, starting sequence at t_zero {}s".format(
+                    tspan, t_zero
+                )
+            )
+            data_snippet = self.cut(tspan=tspan, t_zero=t_zero)
+            data_snippet.set_siq_quantifier(
+                calibration=Calibration(cal_list=[F_mol]), carrier="He"
+            )
+            # now calculate the deconvoluted current based in the signal
+            t_partcurr, v_partcurr = data_snippet.grab_deconvoluted_current(
+                mol=mol,
+                impulse_response=impulse_response,
+                tspan=None,
+                tspan_bg=t_bg,
+                snr=snr,
+            )
+            t_v_list.append((t_partcurr, v_partcurr))
+
+            if plot:
+                axes = data_snippet.plot(
+                    mol_list=[mol], logplot=False, tspan_bg=t_bg, alpha=0.5
+                )
+                axes[1].set_alpha(1)  # this doesnt work, not sure why
```

Comment:
What is this supposed to do?

Reply:
change the transparency of color of whats plotted in axes[1]

Reply:
Ideally I would like this to be applied only to the calibrated, non-deconvoluted molecule line. The way it's done now, also the current and potential are plotted lighter (with alpha = 0.5).

Reply:
In the examples here, `set_alpha` is never called from an Axis object, but from an "artist" object. 
https://www.geeksforgeeks.org/matplotlib-axes-axes-set_alpha-in-python/
I'm not sure how you'd get at the "artist" at this point in the code.

Reply:
I read in the matplotlib documentation that you can try and `set_alpha` on an axis, but it "isnt supported by all backends". https://matplotlib.org/3.2.2/api/_as_gen/matplotlib.axes.Axes.set_alpha.html
I tried all the backends I had easily available and it worked with none of them :P
I'll leave this as a TODO and have future me come up with something, unless you think we need to solve this now?